File: plugin/generators/lib/dist_dir.py
"""dist_dir.py — manage the <deck>/dist/ output directory for the PPTX exporter.
Spec: R012 (output dir hygiene)."""

from pathlib import Path
import shutil

def ensure_dist(deck_path: Path, clean: bool = False) -> Path:
    """Ensure <deck>/dist/ exists. If clean=True, remove it first.
    Returns the absolute Path to the dist dir."""
    dist = Path(deck_path) / "dist"
    if clean and dist.exists():
        shutil.rmtree(dist)
    dist.mkdir(parents=True, exist_ok=True)
    return dist

def ensure_gitignore(deck_path: Path) -> None:
    """Append `dist/` to <deck>/.gitignore if missing. Idempotent."""
    gi = Path(deck_path) / ".gitignore"
    needed = "dist/"
    if gi.exists():
        existing = gi.read_text(encoding="utf-8")
        if needed in existing.splitlines():
            return
        gi.write_text(existing + ("\n" if not existing.endswith("\n") else "") + needed + "\n", encoding="utf-8")
    else:
        gi.write_text(needed + "\n", encoding="utf-8")

def artifact_path(deck_path: Path, name: str) -> Path:
    """Compute deterministic artifact path within <deck>/dist/."""
    return ensure_dist(deck_path) / name

File: plugin/generators/lib/dist_dir.mjs
// dist_dir.mjs — manage <deck>/dist/ for the video recorder + notes.md exports.
// Spec: R012 (output dir hygiene). Mirrors the Python helper API.
import { mkdir, rm, readFile, writeFile, access } from 'node:fs/promises';
import { join } from 'node:path';

export async function ensureDist(deckPath, { clean = false } = {}) {
  const dist = join(deckPath, 'dist');
  if (clean) {
    await rm(dist, { recursive: true, force: true });
  }
  await mkdir(dist, { recursive: true });
  return dist;
}

export async function ensureGitignore(deckPath) {
  const gi = join(deckPath, '.gitignore');
  const needed = 'dist/';
  let existing = '';
  try {
    existing = await readFile(gi, 'utf8');
  } catch { /* doesn't exist yet */ }
  const lines = existing.split(/\r?\n/);
  if (lines.includes(needed)) return;
  const updated = (existing && !existing.endsWith('\n') ? existing + '\n' : existing) + needed + '\n';
  await writeFile(gi, updated, 'utf8');
}

export async function artifactPath(deckPath, name) {
  await ensureDist(deckPath);
  return join(deckPath, 'dist', name);
}
